ros2_batch_job/vendor/README.md

```diff
@@ -14,22 +14,79 @@ This import can be updated by following these steps:
 $ git remote -v
 ```
 
-1. If the upstream (e.g. `https://github.com/osrf/osrf_pycommon.git`) is already remote use that remote name. Otherwise add it as a remote like so (using `osrf_pycommon` as an example):
+2. If the upstream (e.g. `https://github.com/osrf/osrf_pycommon.git`) is already remote use that remote name. Otherwise add it as a remote like so (using `osrf_pycommon` as an example):
 
 ```
 $ git remote add osrf_pycommon_upstream https://github.com/osrf/osrf_pycommon.git
 ```
 
-1. Now run the update command:
+3. Now run the update command:
 
 ```
 $ git subtree pull -P ros2_batch_job/vendor/osrf_pycommon --squash osrf_pycommon_upstream master
 ```
 
+##### When the subtree update fails
+
+When running `git subtree pull` git looks for a commit history in the form of:
+
+```
+    git-subtree-dir: ros2_batch_job/vendor/osrf_pycommon
+    git-subtree-split: <commit-hash>
+```
+
+This can fail for many reasons and if it does it wont udpate the git tree.
+One easy alternative is to remove an re-add the subtree but this is not great
+and laeaves the git subtree history a bit fuzzy. An alternative is to manually
+create the squash merge and add the corresponding commit information so future 
+runs of `git subtree pull` can find it.
```

Comment:
```suggestion
This can fail for many reasons and if it does it won't update the git tree.
One easy alternative is to remove and re-add the subtree but this is not great
and leaves the git subtree history a bit fuzzy.
An alternative is to manually create the squash merge and add the corresponding
commit information so future runs of `git subtree pull` can find it.
```